File: src/test/java/com/polygloat/controllers/LoggedRequestFactory.java
package com.polygloat.controllers;

import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;

public class LoggedRequestFactory {

    static String token;

    static void init(String token) {
        LoggedRequestFactory.token = token;
    }

    public static MockHttpServletRequestBuilder loggedGet(String url) {
        return addToken(MockMvcRequestBuilders.get(url));
    }

    public static MockHttpServletRequestBuilder loggedPost(String url) {
        return addToken(MockMvcRequestBuilders.post(url));
    }

    public static MockHttpServletRequestBuilder loggedDelete(String url) {
        return addToken(MockMvcRequestBuilders.delete(url));
    }

    private static MockHttpServletRequestBuilder addToken(MockHttpServletRequestBuilder builder) {
        return builder.header("Authorization", String.format("Bearer %s", token));
    }
}

File: src/test/kotlin/io/tolgee/fixtures/LoggedRequestFactory.kt
package io.tolgee.fixtures

import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders

object LoggedRequestFactory {
  var token: String? = null

  @JvmStatic
  fun init(token: String?) {
    LoggedRequestFactory.token = token
  }

  @JvmStatic
  fun loggedGet(url: String): MockHttpServletRequestBuilder {
    return addToken(MockMvcRequestBuilders.get(url))
  }

  @JvmStatic
  fun loggedPost(url: String): MockHttpServletRequestBuilder {
    return addToken(MockMvcRequestBuilders.post(url))
  }

  @JvmStatic
  fun loggedPut(url: String): MockHttpServletRequestBuilder {
    return addToken(MockMvcRequestBuilders.put(url))
  }

  @JvmStatic
  fun loggedDelete(url: String): MockHttpServletRequestBuilder {
    return addToken(MockMvcRequestBuilders.delete(url))
  }

  fun addToken(builder: MockHttpServletRequestBuilder): MockHttpServletRequestBuilder {
    return builder.header("Authorization", String.format("Bearer %s", token))
  }
}
